### PR TITLE
chore: remove unnecessary checks for value xfer to system contracts

### LINKED ISSUE #13286: Consistently handle calls with value to system contracts
### Problem

Right now, there is code to surface an error to the caller (`INVALID_FEE_SUBMITTED`) when a call is made to a system contract with value.  Each system contract is responsible to detecting this error condition and responding.  A universal check was later added but the checks inside the system contract still remain.

### Solution

- Re-examine if `INVALID_FEE_SUBMITTED` is the appropriate error code to return to the user.
- Remove checks for value transfer from within system contracts

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
There are checks in some of the system contracts which ensure that value is not passed to the contract.  These checks have been made superfluous by this pr #14843 and thus need to be removed.

**Related issue(s)**:

Fixes #13286

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/ExchangeRateSystemContract.java [hunk 1]
```diff
@@ -63,7 +63,6 @@ public FullResult computeFully(@NonNull Bytes input, @NonNull MessageFrame messa
         requireNonNull(messageFrame);
         try {
             validateTrue(input.size() >= 4, INVALID_TRANSACTION_BODY);
-            validateTrue(messageFrame.getValue().getAsBigInteger().equals(BigInteger.ZERO), INVALID_FEE_SUBMITTED);
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/ExchangeRateSystemContract.java:66]: What about L82-83?  Unneeded now?

Comment (david-bakin-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/ExchangeRateSystemContract.java:66]: And see also `ExchangeRateSystemContractTest.valueShouldNotBeSentToThePrecompile` - doesn't seem still applicable?

**Review by david-bakin-sl — COMMENTED** [review]
And now I'm also wondering if `CustomExceptionalHaltReason.INVALID_FEE_SUBMITTED` is necessary?

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/ExchangeRateSystemContract.java (hunk 1 above)
Comment (lukelee-sl) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/ExchangeRateSystemContract.java:66]: good point.  removed both.

**Review by david-bakin-sl — APPROVED** [review]
LGTM

**Pull request merged**